src/Mailgun/Api/Suppressions/Bounce.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/*
+ * Copyright (C) 2013-2016 Mailgun
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+namespace Mailgun\Api;
+
+use Mailgun\Assert;
+use Mailgun\Resource\Api\Suppressions\Bounce\CreateResponse;
+use Mailgun\Resource\Api\Suppressions\Bounce\DeleteResponse;
+use Mailgun\Resource\Api\Suppressions\Bounce\IndexResponse;
+use Mailgun\Resource\Api\Suppressions\Bounce\ShowResponse;
+
+/**
+ * @see https://documentation.mailgun.com/api-suppressions.html#bounces
+ *
+ * @author Sean Johnson <sean@mailgun.com>
+ */
+class Bounce extends HttpApi
+{
+    use Pagination;
+
+    /**
+     * @return string
+     */
+    protected function getPaginationBase()
+    {
+        return IndexResponse::class;
+    }
+
+    /**
+     * @param string $domain Domain to list bounces for
+     * @param int    $limit  optional
+     *
+     * @return IndexResponse
+     */
+    public function index($domain, $limit = 100)
+    {
+        Assert::notEmpty($domain);
```

Comment:
If `$domain` is `['asdfg']` this assert will not throw an error. We should validate this is a String too.